frontend/catalyst/compiler.py

```diff
@@ -15,15 +15,50 @@
 MLIR/LLVM representations.
 """
 
+import io
 import os
+import sys
 import shutil
 import subprocess
 import warnings
+from io import TextIOWrapper
+from typing import Optional, List
+from dataclasses import dataclass
+from functools import partial
 
 from catalyst._configuration import INSTALLED
 
 package_root = os.path.dirname(__file__)
 
+
+@dataclass
+class CompileOptions:
+    """ Common compile options """
+    verbosity: int
+    logfile: Optional[TextIOWrapper] = None # stdout/stderr or a file
+
+    def get_logfile(self) -> TextIOWrapper:
+        return self.logfile if self.logfile else sys.stderr
+
+default_compile_options:CompileOptions = CompileOptions(0, None)
+
+
+def run_writing_command(command:List[str],
+                        ofile:Optional[str],
+                        compile_options:Optional[CompileOptions] = None) -> None:
+    compile_options: CompileOptions = compile_options if compile_options else default_compile_options
+    runner = partial(subprocess.run, command, check=True)
+    if ofile is not None:
+        if compile_options.verbosity > 0:
+            print(f"[RUNNING] {' '.join(command)} > {ofile}", file=compile_options.get_logfile())
```

Comment:
You can get rid of these `ofile is not None` special case by changing the flags such that instead of writing to stdout you actually add an `-o` file and the name of the outfile as the command.

Reply:
Done